pages/k8s/aws-iam-auth.md

```diff
@@ -0,0 +1,263 @@
+this mapping is called an IAMIdentityMapping and looks something like this:
+
+```yaml
+apiVersion: iamauthenticator.k8s.aws/v1alpha1
+kind: IAMIdentityMapping
+metadata:
+  name: kubernetes-admin
+spec:
+  # Arn of the User or Role to be allowed to authenticate
+  arn: arn:aws:iam::xxxxx:role/k8s-view-role
+  # Username that Kubernetes will see the user as, this is useful for setting
+  # up allowed specific permissions for different users
+  username: john
+  # Groups to be attached to your users/roles. For example `system:masters` to
+  # create cluster admin, or `view` for view only,
+  groups:
+  - view
+```
+
+### Using AWS-IAM with kubectl
+
+#### Download aws-iam-authenticator
+
+The `aws-iam-authenticator` binary needs to be installed on the machine that is
+running kubectl. This is executed by kubectl in order to log in and get a token,
+which is then passed to the Kubernetes API server. You can find the binary
+on the [aws-iam-authenticator releases page][aws-iam-authenticator-releases].
+
+#### Updating kubectl config
+
+In order to use the [aws-iam-authenticator][aws-iam-authenticator-github] with
+kubectl, an updated config file is needed. The config file written to the
+kubernetes-master unit will have a user named `aws-iam-user` that uses the
+`aws-iam-authenticator` client binary and a context named aws-iam-authenticator.
+First, copy the config:
+
+```bash
+juju scp kubernetes-master/0:config ~/.kube/config
+```
+
+The config file will need to be edited in order to add the desired
+Amazon Resource Name(ARN) for authentication. Information about this can
+be found on the
+[aws-iam-authenticator documentation][aws-iam-authenticator-config].
+It is necessary to replace <<insert_arn_here>> with the ARN of the
+aws user or role desired. It may also be necessary to adjust the
+command to fully-qualify the path to `aws-iam-authenticator` if
+it is not located in your path.
+
+The context that uses `aws-iam-authenticator` can be selected with:
+
+```bash
+kubectl config use-context aws-iam-authenticator
+```
+
+Note that the `aws-iam-authenticator` binary behaves like the
+`aws` command line interface in that it will read environment
+variables like AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.
+
+### A note about authorisation
+
+The AWS-IAM charm can be used for authentication only or can be used in an
+RBAC-enabled cluster to authorise users as well. If the charm is related to
+a Charmed Kubernetes cluster without RBAC enabled, any valid AWS IAM
+credential that can assume a role specified in the IAMIdentityMapping
+CRD will be able to run commands against the cluster. If RBAC is enabled,
+the user will have the permissions of the user defined in the
+IAMIdentityMapping CRD.
+
+### Enabling RBAC
+
+In order to get authorisation with AWS-IAM, you will need to use RBAC.
+Refer to the Charmed Kubernetes [RBAC documentation][k8s-rbac-docs] for complete options, but at
```

Comment:
there seems to be a missing word after 'at'.

Reply:
"at minimum you will need..."